modules/calib3d/src/stereosgbm.cpp

```diff
@@ -829,6 +822,636 @@ static void computeDisparitySGBM( const Mat& img1, const Mat& img2,
+    short* disp2Buf;
+    getBufferPointers(cur_buffer,width,width1,D,img1->channels(),SH2,P2,
+                      curCostVolumeLine,hsumBuf,pixDiff,tmpBuf,horPassCostVolume,
+                      vertPassCostVolume,vertPassMin,rightPassBuf,disp2CostBuf,disp2Buf);
+
+    // start real processing:
+    for(int y=src_start_idx;y<src_end_idx;y++)
+    {
+        getRawMatchingCost(curCostVolumeLine,hsumBuf,pixDiff,tmpBuf,y,src_start_idx);
+
+        short* disp_row = (short*)cur_disp.ptr(dst_offset+(y-src_start_idx));
+
+        // initialize the auxiliary buffers for the pseudo left-right consistency check:
+        for(int x=0;x<width;x++)
+        {
+            disp2Buf[x] = (short)INVALID_DISP_SCALED;
+            disp2CostBuf[x] = SHRT_MAX;
+        }
+        CostType* C = curCostVolumeLine - D;
+        CostType prev_min, min_cost;
+        int d, best_d;
+
+        // forward pass
+        prev_min=0;
+        for (int x=D;x<(1+width1)*D;x+=D)
+            accumulateCostsLeftTop(horPassCostVolume+x,horPassCostVolume+x-D,vertPassCostVolume+x,C+x,prev_min,vertPassMin[x/D],D,P1,P2);
+
+        //backward pass
+        memset(rightPassBuf,0,D*sizeof(CostType));
+        prev_min=0;
+        for (int x=width1*D;x>=D;x-=D)
+        {
+            accumulateCostsRight(rightPassBuf,vertPassCostVolume+x,horPassCostVolume+x,C+x,prev_min,D,P1,P2,best_d,min_cost);
+
+            if(uniquenessRatio>0)
+            {
+#if CV_SIMD128
+                horPassCostVolume+=x;
+                int thresh = (100*min_cost)/(100-uniquenessRatio);
+                v_int16x8 thresh_reg = v_setall_s16((short)(thresh+1));
+                v_int16x8 d1 = v_setall_s16((short)(best_d-1));
+                v_int16x8 d2 = v_setall_s16((short)(best_d+1));
+                v_int16x8 eight_reg = v_setall_s16(8);
+                v_int16x8 cur_d(0,1,2,3,4,5,6,7);
+                v_int16x8 mask,cost1,cost2;
+
+                for( d = 0; d < D; d+=16 )
+                {
+                    cost1 = v_load_aligned(horPassCostVolume+d);
+                    cost2 = v_load_aligned(horPassCostVolume+d+8);
+
+                    mask = cost1 < thresh_reg;
+                    mask = mask & ( (cur_d<d1) | (cur_d>d2) );
+                    if( v_signmask(mask) )
+                        break;
+
+                    cur_d = cur_d+eight_reg;
+
+                    mask = cost2 < thresh_reg;
+                    mask = mask & ( (cur_d<d1) | (cur_d>d2) );
+                    if( v_signmask(mask) )
+                        break;
+
+                    cur_d = cur_d+eight_reg;
+                }
+                horPassCostVolume-=x;
+#else
+                for( d = 0; d < D; d++ )
+                {
+                    if( horPassCostVolume[x+d]*(100 - uniquenessRatio) < min_cost*100 && std::abs(d - best_d) > 1 )
+                        break;
+                }
+#endif
+                if( d < D )
+                    continue;
+            }
+            d = best_d;
```

Comment:
The ARM builder produces a warning: 
`stereosgbm.cpp:1383:39: warning: ‘best_d’ may be used uninitialized in this function [-Wmaybe-uninitialized]`